tcms/testruns/tests/test_views.py

```diff
@@ -565,3 +565,157 @@ def test_show_add_cases_to_run(self):
             ]
             for html in html_pieces:
                 self.assertContains(response, html, html=True)
+
+
+class TestCaserunEditMenus(BaseCaseRun):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+    def setUp(self):
+        self.url = reverse('testruns-get', args=[self.test_run.pk])
+
+        self.menu_perms = [
+            'testruns.change_testcaserun',
+            'testruns.add_testcaserun',
+            'testruns.delete_testcaserun',
+            'testcases.add_bug',
+            'testcases.delete_bug',
+            'django_comments.add_comment'
+        ]
+
+        # Setup Permitted User
+
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.tester.username,
+            password='password')
+
+        for perm_string in self.menu_perms:
+            user_should_have_perm(self.tester, perm_string)
+
+        # Setup Unpermitted User
+
+        self.unauthorized = UserFactory()
+        self.unauthorized.set_password('password')
+        self.unauthorized.save()
+
+        for perm_string in self.menu_perms:
+            remove_perm_from_user(self.unauthorized, perm_string)
```

Comment:
It would be best to have permissions assigned inside the test method and only assign/remove permissions for which we are testing. Nothing else! This will mean the tests are in their most strict form and will catch possible errors in the future where we mess up permissions.